src/main/java/com/amalitech/gpuconfigurator/service/attribute/AttributeServiceImpl.java

```diff
@@ -0,0 +1,175 @@
+package com.amalitech.gpuconfigurator.service.attribute;
+
+import com.amalitech.gpuconfigurator.dto.GenericResponse;
+import com.amalitech.gpuconfigurator.dto.attribute.*;
+import com.amalitech.gpuconfigurator.model.attributes.Attribute;
+import com.amalitech.gpuconfigurator.model.attributes.AttributeOption;
+import com.amalitech.gpuconfigurator.repository.attribute.AttributeOptionRepository;
+import com.amalitech.gpuconfigurator.repository.attribute.AttributeRepository;
+import jakarta.persistence.EntityNotFoundException;
+import jakarta.validation.constraints.NotNull;
+import lombok.RequiredArgsConstructor;
+import org.springframework.http.HttpStatus;
+import org.springframework.stereotype.Service;
+
+import java.time.LocalDateTime;
+import java.util.List;
+import java.util.UUID;
+import java.util.stream.Collectors;
+
+@Service
+@RequiredArgsConstructor
+public class AttributeServiceImpl implements AttributeService {
+
+    private final AttributeRepository attributeRepository;
+    private final AttributeOptionRepository attributeOptionRepository;
+
+    @Override
+    public List<AttributeResponse> getAllAttributes() {
+        List<Attribute> attributes = attributeRepository.findAll();
+
+        return attributes.stream()
+                .map(this::createAttributeResponseType)
+                .collect(Collectors.toList());
+    }
+
+    @Override
+    public Attribute addAttribute(@NotNull AttributeDto attribute) {
+        Attribute newAttribute =  Attribute.builder()
+                .attributeName(attribute.attributeName())
+                .build();
+
+        return attributeRepository.save(newAttribute);
+    }
+
+    @Override
+    public AttributeResponse updateAttribute(UUID id, @NotNull AttributeDto attribute) {
+        Attribute updateAttribute = attributeRepository.findById(id).orElseThrow(() -> new EntityNotFoundException("attribute not found"));
+
+        updateAttribute.setAttributeName(attribute.attributeName());
+        updateAttribute.setUpdatedAt(LocalDateTime.now());
+
+        Attribute savedAttribute = attributeRepository.save(updateAttribute);
+        return this.createAttributeResponseType(savedAttribute);
+   }
+
+   @Override
+   public Attribute getAttributeByName(String name) {
+       return attributeRepository.findByAttributeName(name).orElseThrow(() -> new EntityNotFoundException("attribute not found"));
+   }
+
+    @Override
+    public AttributeResponse getAttributeById(UUID id) {
+        Attribute attribute = attributeRepository.findById(id).orElseThrow(() -> new EntityNotFoundException("attribute not found"));
```

Comment:
The attribute not found message has been replicated about 3 times now, might be nice to have it at one place to aid with maintainability

Reply:
done